Spatial_searching/test/Spatial_searching/remove.cpp

```diff
@@ -0,0 +1,29 @@
+#include <CGAL/Simple_cartesian.h>
+#include <CGAL/Search_traits_2.h>
+#include <CGAL/Kd_tree.h>
+
+typedef CGAL::Simple_cartesian<double> K;
+typedef K::Point_2 Point;
+typedef CGAL::Search_traits_2<K> Traits;
+typedef CGAL::Kd_tree<Traits> Tree;
+
+int main()
+{
+  Tree t;
+  t.insert(Point(0,0));
+  t.insert(Point(1,2));
+  t.insert(Point(2,0));
+  t.remove(Point(1,2));
+  t.insert(Point(3,4));
+  t.remove(Point(0,0));
+  t.remove(Point(3,4));
+  t.insert(Point(5,5));
+  t.build();
+  assert(t.size()==2);
+  t.clear();
+  for(int i=0;i<1000;++i)
+    t.insert(Point(i,-i));
+  for(int i=0;i<1000;++i)
+    t.remove(Point(i,-i));
+  t.print();
```

Comment:
This test should use `assert`, to check something (like the number of remaining points in the tree, and the closest point of a known query) because, in the testsuite, a few columns of the testsuite compile and run the tests with `CGAL_NDEBUG`.